_components/input-mask/guidance/implementation.md

```diff
@@ -1,24 +1,84 @@
+- **Add a `maxlength` attribute to the input element.** This will be used as the limit referenced in the message and for validation.
+- **Set the attribute `type` to numeric if your mask is all numbers.** This will display the numeric keypad instead of the full keyboard.
+
+{:.usa-content-list }
+The following is an example of `input` element attributes for a Social Security Number mask:
 
 {:.usa-content-list }
 ```
-placeholder="___ ___ ___"
-pattern="\d\d\d \d\d\d \d\d\d"
+aria-describedby="ssn-hint ssn-hint-info"
+class="usa-input usa-input-mask__field"
+id="ssn"
 inputmode="numeric"
+mask="999-99-9999"
+maxlength="11"
+name="ssn"
+placeholder="___-__-____"
+type="text"
 ```
 
 {:.usa-content-list }
 <h3>The input mask and alphanumeric characters</h3>
 
 {:.usa-content-list }
-Input mask examples include an alphanumeric version which can be customized to meet your needs. To implement an alphanumeric input mask, you will need to add the attribute `data-charset` to the input tag. Similar to the `pattern` attribute, include a number sign `#` for each number and an `A` for each required letter. We recommend changing inputmode to `text` so the full keyboard is available to your user.
+Input mask examples include an alphanumeric version which can be customized to meet your needs. Include the number `9` for every number and a capitalized letter `A` for every required letter. We recommend changing inputmode to `text` so the full keyboard is available to your user.
 
 {:.usa-content-list }
-If your input mask calls for a pattern such as _1A2 B3C_, you would implement that by adding the following attributes and character formatting to your input element:
+If your input mask calls for a complicated pattern like _1EG4-TE5-MK73_, you could implement that by adding the following attributes and character formatting to your `input` element:
 
 {:.usa-content-list }
 ```
-placeholder="___ ___"
-pattern="\w\d\w \d\w\d"
-data-charset="A#A #A#"
+aria-describedby="alphanumeric-hint alphanumeric-hint-info"
+class="usa-input usa-input-mask__field"
+forceupper="true"
+id="alphanumeric"
+mask="9AA9-AA9-AA99"
+name="alphanumeric"
+placeholder="____-___-____"
+maxlength="13"
+type="text"
 ```
+
+{:.usa-content-list }
+<h3>Additional parameters</h3>
+
+{:.usa-content-list }
+Additional parameters can be used to force uppercase or lowercase characters. Add `forceupper` or `forcelower` to the `input` element and set either of them to true to force this behavior.
+
+{:.usa-content-list }
+<h3>Error handling</h3>
+
+{:.usa-content-list }
+The input mask utilizes screen reader-friendly logic to gently notify a user if the character they typed is incorrect. The Preview section above illustrates this.
+
+{:.usa-content-list }
+<h3>Available format characters</h3>
+
+{:.usa-content-list }
+Here is a complete list of format characters that can be used in your implementation of input mask:
+
+{:.usa-content-list }
```

Comment:
Can this table match the format of the other tables (for settings, variants, properties, etc) on our guidance pages? They use at least the `usa-table--borderless site-table-responsive site-table-simple` classes as well as `font-lang-3xs`, among others.

Reply:
@amyleadem check the latest for all the updates... Looks so much better, thanks for all suggestions!
https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/mm-input-mask-guidance/components/input-mask/